File: swift/HyperBabelDemo/API/PushService.swift
/*
 * HyperBabel API — push notification token management.
 */
import Foundation

struct PushToken: Codable, Identifiable {
    var id: String { token }
    let token: String
    let platform: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case token, platform
        case createdAt = "created_at"
    }
}

struct PushTokenList: Codable {
    let tokens: [PushToken]?
}

enum PushService {
    static func register(userId: String, token: String, platform: String = "ios") async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "POST", "/push/register",
            body: ["user_id": userId, "token": token, "platform": platform]
        )
    }

    static func unregister(userId: String, token: String) async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "DELETE", "/push/unregister",
            body: ["user_id": userId, "token": token]
        )
    }

    static func getTokens(userId: String) async throws -> PushTokenList {
        try await ApiClient.shared.request(
            "GET", "/push/tokens",
            query: [("user_id", userId)]
        )
    }
}

File: kotlin/app/src/main/kotlin/com/hyperbabel/demo/api/PushService.kt
/*
 * HyperBabel API — push notification token management.
 */
package com.hyperbabel.demo.api

import kotlinx.serialization.json.JsonObject
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.HTTP
import retrofit2.http.POST
import retrofit2.http.Query

interface PushService {
    @POST("api/v1/push/register")
    suspend fun registerToken(@Body body: Map<String, String>)

    @HTTP(method = "DELETE", path = "api/v1/push/unregister", hasBody = true)
    suspend fun unregisterToken(@Body body: Map<String, String>)

    @GET("api/v1/push/tokens")
    suspend fun getTokens(@Query("user_id") userId: String): JsonObject
}
